protocol/eth-shared-lockbox.md

```diff
@@ -0,0 +1,120 @@
+
+- The initial migrated ETH, serves as the starting amount.
+- A deposit mapping that tracks ETH successfully deposited and finalized withdrawals.
+- The `InteropETHTransferred` mapping is implemented when `SuperchainWETH` interoperability transfers are enabled.
+
+Then, the opt-out process would be performed as follows:
+
+1. Disconnect the chain from the interop set.
+2. Pause new deposits.
+3. Waiting time for stabilizing pending interop messages and deposits.
+4. Trigger the migration by porting `InteropETHTransferred` back to L1.
+
+One path could be having two functions to initiate and finalize the migration, or directly via a protocol upgrade.
+
+The resultant math to calculate the exact ETH amount to migrate will be:
+
+ETH to Migrate = (1)initial balance + (2)deposit mapping + (3)Netflow ETH interop transferred.
+Note that (1) should be zero if the chain is deployed using `SharedLockbox` by default. (2) and (3) are integers as their number could become negative. For (3) to be an accurate value, there should be no pending messages waiting to expire. 
+
+### Impact
+
+The whole work can be split into two big phases:
+
+1. **First phase**: Modify the necessary core L1 contracts, adding the `InteropETHTransferred` counter for `SuperchainWETH` total transfer history and the standardized process for new and existing chains to adopt the `SharedLockbox`.
+2. **Second phase**: Develop the needed pieces for the opt-out process.
+
+All phases require an audit of the new and modified contracts: `OptimismPortal`, `SharedLockbox`, `SystemConfigInterop`, `SuperchainWETH` transfer mapping in L2, and possibly others. This also includes the scripts to perform the migration.
+
+# Alternatives Considered
+
+### Reverts on L2
+
+One alternative is not to share liquidity at the L1 level and instead prevent withdrawals from being stuck by reverting them on L2. This approach requires tracking the exact ETH balance in L2 via deposits and withdrawals. An `ETHBalance` counter would increment with new mints from deposits and decrease with successful `initiateWithdrawal` calls.
+
+This method would require minimal changes to `L1Block` and adjustments to how `TransactionDeposited` is processed to validate new ETH counts. Additionally, it necessitates porting the initial balance as part of an upgrade transaction.
+
+The problem with L2 reverts is that it breaks the ETH withdrawal guarantee invariant and still exposes the system to solvency issues. Based on previous feedback, this would affect existing applications and pre-deployed contracts such as the current `FeeVault` withdrawal contract logic.
```

Comment:
This logic would be implemented in the `L2ToL1MessagePasser` in practice, it's the lowest level abstraction. I am confident in being able to get the "start balance" into L2, but I do agree that it would break applications that assume they can withdraw, but at least their ether wouldn't be stuck